packages/components/src/components/hds/disclosure-primitive/index.ts

```diff
@@ -33,11 +35,44 @@ export interface HdsDisclosurePrimitiveSignature {
 }
 
 export default class HdsDisclosurePrimitiveComponent extends Component<HdsDisclosurePrimitiveSignature> {
-  @tracked isOpen = this.args.isOpen ?? false;
+  @tracked _isOpen = false;
+  @tracked _isControlled = this.args.isOpen != undefined;
+
+  get isOpen() {
+    if (this._isControlled) {
+      // if the state is controlled from outside, the argument overrides the internal state
+      return this.args.isOpen ?? this._isOpen;
+    } else {
+      // if the state is changes internally, the internal state overrides the argument
+      return this._isOpen !== undefined
+        ? this._isOpen
+        : Boolean(this.args.isOpen);
+    }
+  }
+
+  set isOpen(value) {
+    this._isOpen = value || false;
+  }
 
   @action
   onClickToggle() {
     this.isOpen = !this.isOpen;
+    this._isControlled = false;
+    // we call the "onClickToggle" callback if it exists and it's a function
+    if (
+      this.args.onClickToggle &&
+      typeof this.args.onClickToggle === 'function'
+    ) {
+      this.args.onClickToggle(this.isOpen);
+    }
+  }
+
+  @action
+  onStateChange() {
+    if (this.args.isOpen !== undefined) {
```

Comment:
if I understand correctly, we don't use `this._isControlled` here because this function is invoked when the value of `@isOpen` changes, which triggers the `did-update` which is run without re-rendering the component, so line 39 is not re-evaluated

if that is the case, what do you think of adding a comment to explain this "detail"? 😀 (it took me a while to get it)

if not, then maybe we can use `this._isControlled` as condition? would make sense (in which case TS complains, but we can cast the assignment via `Boolean(this.args.isOpen)` like we do at line 49